packages/foam-core/src/utils/hashtags.ts

```diff
@@ -1,16 +1,21 @@
 import { isSome } from './core';
-const HASHTAG_REGEX = /(^|\s)#([0-9]*[\p{L}/_-][\p{L}\p{N}/_-]*)/gmu;
-const WORD_REGEX = /(^|\s)([0-9]*[\p{L}/_-][\p{L}\p{N}/_-]*)/gmu;
+const HASHTAG_REGEX = /(?<=^|\s)#([0-9]*[\p{L}/_-][\p{L}\p{N}/_-]*)/gmu;
```

Comment:
the lookbehind makes the matching more accurate, it doesn't include the previous char on the match, which makes trimming unnecessary, and the `index` reliable